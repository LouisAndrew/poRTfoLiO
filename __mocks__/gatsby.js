const React = require('react')
const gatsby = jest.requireActual('gatsby')

const mockSocials = [
    { link: 'http://localhost', social_key: 'github' },
    { link: 'http://google.com', social_key: 'dribbble' },
]

const mockButtonText = 'Click me'
const mockContactContent = 'Content'
const mockContactHeadline = 'Headline'
const mockEmail = 'john@doe.com'

module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to,
            })
    ),
    StaticQuery: jest.fn(),
    // can I mock return value here?
    useStaticQuery: jest.fn(() => ({
        socialData: {
            frontmatter: {
                social: mockSocials,
            },
        },
        emailData: {
            frontmatter: {
                contact: {
                    button_text: mockButtonText,
                    contact_content: mockContactContent,
                    contact_headline: mockContactHeadline,
                    email: mockEmail,
                },
            },
        },

        // forced to write the hero-page-query mock here.
        heroData: {
            frontmatter: {
                hero: {
                    hero_content: 'Hello, World!', // hard coded from hero.test.tsx
                    hero_headline: 'I am headline', // Also hard coded from hero.test.tsx
                },
            },
        },
        imgS: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgM: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
        imgXL: {
            frontmatter: {
                hero: {
                    hero_imgs: [
                        {
                            childImageSharp: {
                                fixed: {},
                            },
                        },
                    ],
                },
            },
        },
    })),
}
