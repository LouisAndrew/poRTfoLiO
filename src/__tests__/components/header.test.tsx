import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

// import { render, cleanup } from '@testing-library/react'
import { cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Header from 'components/header'

import withContext from 'helper/utils/with-context'

import * as Gatsby from 'gatsby'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

describe('Header', () => {
    const Element = withContext({
        children: <Header />,
    })

    afterEach(cleanup)
    beforeEach(() => {
        useStaticQuery.mockImplementationOnce(() => ({
            logoImg: {
                frontmatter: {
                    hero: {
                        hero_imgs: [
                            {
                                childImageSharp: {
                                    fixed: {},
                                },
                            },
                        ],
                    },
                },
            },
        }))
    })

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(Element, div)
    })

    /* it('renders correctly', () => {
		const { getByTestId } = render()
	}) */

    it('matches snapshot', () => {
        const run = true

        expect(run).toBeTruthy()

        if (run) {
            const tree = renderer.create(Element).toJSON()
            expect(tree).toMatchSnapshot()
        }
    })
})
