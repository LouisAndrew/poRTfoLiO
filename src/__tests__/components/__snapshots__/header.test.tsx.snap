// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<header
  className="css-q52vjd"
>
  <div
    className="css-1770sze"
  >
    <div
      className="css-1tqp1zr"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "height": undefined,
            "overflow": "hidden",
            "position": "relative",
            "width": undefined,
          }
        }
      >
        <picture>
          <source />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      className="css-16vu25q"
    />
    <div
      className="links-and-theme css-vpi35"
    >
      <div
        className="close-button css-azuqvi"
        onClick={[Function]}
        role="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
          />
        </svg>
      </div>
      <nav
        className="css-azs354"
        id="links"
      >
        <a
          href="/#hero"
          id="header-link-hero"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          HOME
        </a>
        <a
          href="/#projects"
          id="header-link-projects"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          MY PROJECTS
        </a>
        <a
          href="/#about"
          id="header-link-about"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          ABOUT ME
        </a>
        <a
          href="#contact"
          id="header-link-contact"
        >
          CONTACT
        </a>
      </nav>
      <div
        className="css-wtwfv3"
        onClick={[Function]}
        role="button"
      >
        <div
          className="css-9pwli7"
        >
          <svg
            data-testid="button-dark"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M632 888H392c-4.4 0-8 3.6-8 8v32c0 17.7 14.3 32 32 32h192c17.7 0 32-14.3 32-32v-32c0-4.4-3.6-8-8-8zM512 64c-181.1 0-328 146.9-328 328 0 121.4 66 227.4 164 284.1V792c0 17.7 14.3 32 32 32h264c17.7 0 32-14.3 32-32V676.1c98-56.7 164-162.7 164-284.1 0-181.1-146.9-328-328-328zm127.9 549.8L604 634.6V752H420V634.6l-35.9-20.8C305.4 568.3 256 484.5 256 392c0-141.4 114.6-256 256-256s256 114.6 256 256c0 92.5-49.4 176.3-128.1 221.8z"
            />
          </svg>
        </div>
        <p
          className="css-we7pb2"
        >
          Go dark
        </p>
      </div>
    </div>
    <div
      className="open-button css-5sakv7"
      onClick={[Function]}
      role="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </div>
  </div>
</header>
`;
